sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/DataMovementSharesExtensions.cs

```diff
@@ -123,14 +122,14 @@ public static FileSmbProperties GetFileSmbProperties(
             }
             return new()
             {
-                FileAttributes = (options?.IsNfs ?? false)
+                FileAttributes = ((options?.ShareProtocol ?? ShareProtocols.Smb) == ShareProtocols.Nfs)
                     ? default
                     : GetPropertyValue<NtfsFileAttributes>(
                         options?._isFileAttributesSet ?? false,
                         options?.FileAttributes,
                         properties?.RawProperties,
                         DataMovementConstants.ResourceProperties.FileAttributes),
-                FilePermissionKey = (options?.IsNfs ?? false)
+                FilePermissionKey = ((options?.ShareProtocol ?? ShareProtocols.Smb) == ShareProtocols.Nfs)
                     ? default
                     : permissionKeyValue,
```

Comment:
Now that this is an enum, we can make these prettier and more readable. Inline switch statements do work properly with nullable enums.

Currently, this null-coalesces into value A, then checks equality for value B, as a check for the negative case of not setting anything. A bit hard to parse mentally.

Instead, we should be switching on the nullable enum and setting values based on the case.

```csharp
FilePermissionKey = options?.ShareProtocol switch
{
    ShareProtocols.Nfs => default,
    _ => permissionKeyValue, // default case catches both SMB and no value, you could also explicitly define both if desired
},
```